### DISCUSSION TITLE
Make "Enable Low Network Buffers" active when latency is above 64 samples?

### DISCUSSION THREAD

**Discussion by 38github** [jamulussoftware/jamulus#2111] [Ideas]
Is it possible to make Enable Low Network Buffers being active when latency is above 64 samples? If so, why isn't it already?

**Comment by gilgongo** [comment]
I believe the reason why this is a manual thing is that there's some tradeoffs with it. You might well experience worse sound in fact so it's for experimentation (assuming the server has it on of course).

That said, if we think that in **most** cases servers have it on, and in **most** cases its effect will be positive, I guess we could kick it in above 64 samples? It's all an inexact science really.

**Comment by jujudusud** [comment]
Small network buffers is only active when the buffer is set to 64 samples for the client and the option activated for the server.
it is not necessary to make it functional for buffers larger than 64 samples since in this case, to reduce latency, it is enough to reduce the buffer size.

**Reply by 38github** [reply]
Wrong. It is necessary to reduce latenc and reducing the buffer size in Jamulus is not enough.

128/2 (JACK)and 2/2 local/server buffers gives 17ms overall delay
64/4 (JACK) and 2/2 local/server buffers gives 11ms overall delay

https://sourceforge.net/p/llcon/discussion/software/thread/0eddbf5a02/

**Reply by jujudusud** [reply]
With a USB audio interface, it is commonly admited that we have to use 3 periods/buffers:
https://wiki.linuxaudio.org/wiki/list_of_jack_frame_period_settings_ideal_for_usb_interface

**Reply by 38github** [reply]
That is not of interest talking about 64 vs 128 samples. User whatever buffers you want but as long as 64 samples are used it achieves much lower overall delay.

**Reply by dcorson-ticino-com** [reply]
I think the supposition was that if the connection is too poor to support using a 64 sample buffer size it will also not support going even smaller especially considering the accompanying substantial increase in bandwidth necessary for small buffers.

**Reply by jujudusud** [reply]
> That is not of interest talking about 64 vs 128 samples. User whatever buffers you want but as long as 64 samples are used it achieves much lower overall delay.

That is exactly what I wrote in the first comment but you seemed to think that it is wrong.
